First draft content for the Create Expectations: Overview page.

Rachel-Reverie opened the pull request:
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- Adds content for the Create Expectations: Overview page.


After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-


### Definition of Done
Please delete options that are not relevant.

- [X] I have made corresponding changes to the documentation



Thank you for submitting!

Rachel-Reverie committed:
First draft content for the Create Expectations: Overview page.

Shinnnyshinshin commented on docs/guides/expectations/create_expectations_overview.md:
```diff
@@ -3,4 +3,67 @@ title: "Create Expectations: Overview"
+<UniversalMap setup='inactive' connect='inactive' create='active' validate='inactive'/>
+
+:::note Prerequisites
+- Completing [Step 3: Create Expectations](/docs/tutorials/getting_started/create_your_first_expectations) of the Getting Started tutorial is recommended.
```
I think you meant to say `Step 2: Connect To Data` ? :)

Rachel-Reverie replied:
I very much did.  Thanks for the catch.

Rachel-Reverie replied:
Wait, wait -- backtracking.  It should be Step 3: Create Expectations in the tutorial.  We want them to complete the relevant tutorial steps before diving into the docs for more info, since this is the overview for step 3, we want them to have the context from completing step 3 in the tutorial before they try to wade through this.

Shinnnyshinshin commented on docs/guides/expectations/create_expectations_overview.md:
```diff
@@ -3,4 +3,67 @@ title: "Create Expectations: Overview"
+
+There are a few workflows you can potentially follow when creating Expectations.  These workflows represent various ways of creating Expectations, although they converge in the end when you will save and test those Expectations.  Of the three potential ways to create Expectations, two are recommended in particular.
+
+First is the interactive work flow.  In this workflow, you will be working in a Python interpreter or Jupyter notebook.  You will use a Validator and it's `expect()` method to define Expectations in an Expectation Suite, and when you have finished you will save that Expectation Suite into your Expectation Store.  A more thorough overview of this workflow, and a link to an in depth guide on it, ca ...
```
```suggestion
First is the interactive work flow.  In this workflow, you will be working in a Python interpreter or Jupyter notebook.  You will use a Validator and its `expect()` method to define Expectations in an Expectation Suite, and when you have finished you will save that Expectation Suite into your Expectation Store.  A more thorough overview of this workflow, and a link to an in depth guide on it, can be found in this document's section on [creating Expectations interactively](#creating-expectations-automatically).
```

Rachel-Reverie committed:
Update docs/guides/expectations/create_expectations_overview.md

Co-authored-by: William Shin <will@superconductive.com>

Shinnnyshinshin commented on docs/guides/expectations/create_expectations_overview.md:
````diff
@@ -3,4 +3,67 @@ title: "Create Expectations: Overview"
+
+It may be that you have saved an Expectation Suite that you wish to go back to and edit.  The simplest way to do this is to use the CLI.  You can use the command:
+
+```markdown title="Terminal command"
+great_expectations suite edit NAME_OF_YOUR_SUITE_HERE
````
just learned about `markdown title` thank you @Rachel-Reverie

Rachel-Reverie replied:
Happy to have introduced you to it!  I intend for one of our documentation standards to be indicating if the following lines are intended for a terminal or python interpreter, just to try and help people who are skimming the documents or who have gotten slightly lost when context switching between the doc and their working environment.

Shinnnyshinshin reviewed (commented):
HI @Rachel-Reverie love the work so far, I think one question I have is regarding terminology. How are you thinking of differentiating between 1) GE-terms (like batch and Expectations) that are just going to be mentioned in-line and 2) GE-terms that are going to be linked to their respective Reference sections? 

I think there is a balance between the two, since it would also be weird to have the most of the document be pink (ie. links out to the reference sections).  

And please let me know if you would like to keep this PR up, or if you would like an approval. I'm happy either way

Rachel-Reverie commented:
> HI @Rachel-Reverie love the work so far, I think one question I have is regarding terminology. How are you thinking of differentiating between 1) GE-terms (like batch and Expectations) that are just going to be mentioned in-line and 2) GE-terms that are going to be linked to their respective Reference sections?
> 
> I think there is a balance between the two, since it would also be weird to have the most of the document be pink (ie. links out to the reference sections).
> 
> And please let me know if you would like to keep this PR up, or if you would like an approval. I'm happy either way

The plan is for all Technical Terms to be displayed in Title Case when they are inline in the documentation -- but only the first occurrence in the document will receive the treatment where they have a link to an in-depth page and a tooltip with their definition.  (But that will be done in the Glossary update, which is why you don't see those tags in this PR.)

Please leave this PR up.  It goes to the universal-map-docs-update branch, so I can push it through without an approval -- therefore, I'd like to give it some time for a few more people (particularly @talagluck, who helped me build its framework) to review it for potential inaccuracies.

Thank you!

talagluck commented on docs/guides/expectations/create_expectations_overview.md:
```diff
@@ -3,4 +3,67 @@ title: "Create Expectations: Overview"
+
+There are a few workflows you can potentially follow when creating Expectations.  These workflows represent various ways of creating Expectations, although they converge in the end when you will save and test those Expectations.  Of the three potential ways to create Expectations, two are recommended in particular.
+
+First is the interactive work flow.  In this workflow, you will be working in a Python interpreter or Jupyter notebook.  You will use a Validator and its `expect()` method to define Expectations in an Expectation Suite, and when you have finished you will save that Expectation Suite into your Expectation Store.  A more thorough overview of this workflow, and a link to an in depth guide on it, can ...
```
Hmm - calling it the `expect()` method isn't quite accurate. Maybe something like the below? (It's a little clunky, so feel free to adjust).
```suggestion
First is the interactive workflow.  In this workflow, you will be working in a Python interpreter or Jupyter notebook.  You will use a Validator and call expectations as methods on it to define Expectations in an Expectation Suite, and when you have finished you will save that Expectation Suite into your Expectation Store.  A more thorough overview of this workflow, and a link to an in depth guide on it, can be found in this document's section on [creating Expectations interactively](#creating-expectations-automatically).
```

talagluck commented on docs/guides/expectations/create_expectations_overview.md:
```diff
@@ -3,4 +3,67 @@ title: "Create Expectations: Overview"
+
+First is the interactive work flow.  In this workflow, you will be working in a Python interpreter or Jupyter notebook.  You will use a Validator and its `expect()` method to define Expectations in an Expectation Suite, and when you have finished you will save that Expectation Suite into your Expectation Store.  A more thorough overview of this workflow, and a link to an in depth guide on it, can ...
+
+The second workflow is the automated work flow.  In this workflow, you will use a Profiler to generate Expectations based on some input data.  The Validator will then save that those Expectations as an Expectation Suite in an Expectation Store.  You will also have the option of vieiwing them in your Data Docs.  A more thorough overview of this workflow, and a link to an in depth guide on it, can  ...
```
Also, are you set on "automated" workflow? I think it might be simpler to refer to it as the "profiler workflow", especially as we continue to do more work on profilers.
 
```suggestion
The second workflow is the automated workflow.  In this workflow, you will use a Profiler to generate Expectations based on some input data.  The Validator will then save those Expectations as an Expectation Suite in an Expectation Store.  You will also have the option of viewing them in your Data Docs.  A more thorough overview of this workflow, and a link to an in-depth guide on it, can be found in this document's section on [creating Expectations automatically](#creating-expectations-automatically).
```

talagluck commented on docs/guides/expectations/create_expectations_overview.md:
```diff
@@ -3,4 +3,67 @@ title: "Create Expectations: Overview"
+
+Creating Expectations is an integral part of Great Expectations.  By the end of this step, you will have created an Expectation Suite containing one or more Expectations which you will use when you Validate data.
+
+## The Create Expectations process
```
Love this introductory section! Could we add some bolding or other formatting to key terms (especially terms like "interactive workflow")? It might help break up the section a bit.

Rachel-Reverie replied:
I added bold to each of the three types of workflows, so one bold statement near the start of each paragraph to highlight what that paragraph would be about.

talagluck commented on docs/guides/expectations/create_expectations_overview.md:
```diff
@@ -3,4 +3,67 @@ title: "Create Expectations: Overview"
+
+The second workflow is the automated work flow.  In this workflow, you will use a Profiler to generate Expectations based on some input data.  The Validator will then save that those Expectations as an Expectation Suite in an Expectation Store.  You will also have the option of vieiwing them in your Data Docs.  A more thorough overview of this workflow, and a link to an in depth guide on it, can  ...
+
+The third workflow, which is not recommended, is to manually define your Expectations by writing their configurations.  This workflow does not require source data to work against, but does require a deep understanding of the configurations available for Expectations.  We will forgo discussion of it in this document, and focus on the two recommended workflows.  If for some reason you must use this ...
```
The phrase "not recommended" seems a bit harsh. I do like the qualification as a reason for why you don't get into it in this doc. Maybe instead,
```suggestion
The third workflow, which is for advanced users, is to manually define your Expectations by writing their configurations.  This workflow does not require source data to work against, but does require a deep understanding of the configurations available for Expectations.  We will forgo discussion of it in this document, and focus on the two recommended workflows.  If for some reason you must use this workflow, we do provide an in depth guide to it in our documentation on [how to create and edit expectations based on domain knowledge without inspecting data directly](./how_to_create_and_edit_expectations_based_on_domain_knowledge_without_inspecting_data_directly.md).
```

Rachel-Reverie replied:
I like that change.  Less harsh, as you said.

talagluck commented on docs/guides/expectations/create_expectations_overview.md:
```diff
@@ -3,4 +3,67 @@ title: "Create Expectations: Overview"
+
+The third workflow, which is not recommended, is to manually define your Expectations by writing their configurations.  This workflow does not require source data to work against, but does require a deep understanding of the configurations available for Expectations.  We will forgo discussion of it in this document, and focus on the two recommended workflows.  If for some reason you must use this ...
+
+Regardless of which of the first two workflows you have followed, once you have saved your Expectation Suite it is advised that you test them by validating them against the batch or batches of data you created them against.  This process is the same in either workflow, since at this point you will be using a saved Expectation Suite and each of the prior workflows ends with the saving of your Expe ...
```
Since I recommend a slight change to the language about the third method above, it might make sense to change this as well.

```suggestion
When following one the first two workflows, once you have saved your Expectation Suite it is advised that you test it by validating your Expectations against the batch or batches of data against which you created them.  This process is the same in either workflow, since at this point you will be using a saved Expectation Suite and each of the prior workflows ends with the saving of your Expectation Suite.  Instructions for this will be detailed in this document's section on [testing your Expectation Suite](#testing-your-expectation-suite).
```

talagluck commented on docs/guides/expectations/create_expectations_overview.md:
```diff
@@ -3,4 +3,67 @@ title: "Create Expectations: Overview"
+
+### Creating Expectations interactively
+
+When using the interactive method of creating Expectations, you will start as you always do with your Data Context.  In this case, you will want to navigate to your Data Context's root directory in your terminal, where you will use the CLI to launch a Jupyter Notebook which will contain scaffolding to assist you in the process.  You can even provide flags such as `--profile` which will allow you  ...
```
I may have misspoken yesterday. The `--profile` flag will actually push the user to the automatic/profiling workflow below. I believe the options are `great_expectations suite new --interactive` for interactive, and `great_expectations suite new --interactive --profile`  for profiling.

talagluck commented on docs/guides/expectations/create_expectations_overview.md:
```diff
@@ -3,4 +3,67 @@ title: "Create Expectations: Overview"
+
+### Creating Expectations automatically
+
+As with creating Expectations interactively, you will start with your Data Context.  However, in this case you will be working in a Python environment, so you will need to loading or create your Data Context as an instantiated object.  Next, you will create a Batch Request to specify the data you would like to profile in order to create your Expectation Suite.  Once you have a Batch Request confi ...
```
See above. The CLI can direct the user down this path as well.

talagluck commented on docs/guides/expectations/create_expectations_overview.md:
```diff
@@ -3,4 +3,67 @@ title: "Create Expectations: Overview"
+
+When using the interactive method of creating Expectations, you will start as you always do with your Data Context.  In this case, you will want to navigate to your Data Context's root directory in your terminal, where you will use the CLI to launch a Jupyter Notebook which will contain scaffolding to assist you in the process.  You can even provide flags such as `--profile` which will allow you  ...
+
+We provide an in depth guide to using the CLI (and what flags are available to you) for interactively creating Expectations in our guide on [how to create and edit Expectations with instant feedback from a sample batch of data](./how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data.md).
```
This led me down a rabbit hole! It looks like "in-depth" is an adjective, while "in depth" is an adverb, so I think you want the former here (and in a few other places in the doc).

Rachel-Reverie replied:
Correction made.  I think I'm going to have to add something as a "Documentation standards specific to me" to check for that throughout the docs, because it is likely I've overlooked it elsewhere.

talagluck commented on docs/guides/expectations/create_expectations_overview.md:
```diff
@@ -3,4 +3,67 @@ title: "Create Expectations: Overview"
+
+The Profiler we recommend using for this workflow is the `UserConfigurableProfiler`.  We provide an in depth guide to this in our documentation on [how to create and edit expectations with a profiler](./how_to_create_and_edit_expectations_with_a_profiler.md).
+
+We also provide an in depth guides on [how to create a new Expectation Suite using rule based Profilers](./advanced/how_to_create_a_new_expectation_suite_using_rule_based_profilers.md) and [how to create a new Expectation Suite by profiling from a json schema file](./advanced/how_to_create_a_new_expectation_suite_by_profiling_from_a_jsonschema_file.md) in our advanced how to guides for Expectatio ...
```
```suggestion
We also provide in depth guides on [how to create a new Expectation Suite using rule based Profilers](./advanced/how_to_create_a_new_expectation_suite_using_rule_based_profilers.md) and [how to create a new Expectation Suite by profiling from a json schema file](./advanced/how_to_create_a_new_expectation_suite_by_profiling_from_a_jsonschema_file.md) in our advanced how to guides for Expectations.
```

talagluck commented on docs/guides/expectations/create_expectations_overview.md:
```diff
@@ -3,4 +3,67 @@ title: "Create Expectations: Overview"
+
+### Testing your Expectation Suite
+
+Once you have created your Expectation Suite and saved it, you may wish to test it.  The simplest way to do this is to Validate some data against it.  You can do this using a SimpleChekpoint as demonstrated in the [optional step on running validation, saving your suite, and building Data Docs](./how_to_create_and_edit_expectations_with_a_profiler.md#6-optional-running-validation-saving-your-suite ...
```
```suggestion
Once you have created your Expectation Suite and saved it, you may wish to test it.  The simplest way to do this is to Validate some data against it.  You can do this using a SimpleCheckpoint as demonstrated in the [optional step on running validation, saving your suite, and building Data Docs](./how_to_create_and_edit_expectations_with_a_profiler.md#6-optional-running-validation-saving-your-suite-and-building-data-docs) of our [how to create and edit Expectations with a Profiler](./how_to_create_and_edit_expectations_with_a_profiler.md) documentation.  Or you can just move on to [Step 4: Validate Data.](../validation/validate_data_overview.md)
```

talagluck commented on docs/guides/expectations/create_expectations_overview.md:
````diff
@@ -3,4 +3,67 @@ title: "Create Expectations: Overview"
+great_expectations suite edit NAME_OF_YOUR_SUITE_HERE
+```
+
+This will open a Jupyter Notebook that contiains the configurations for each of that Expectation Suite's Expectations in their own cells.  You can edit these cells and then run them to generate Expectations in a new Expectation Suite.  Once your edited version of the Expectations have been created in their own Expectation Suite, you can save that Expectation Suite over the pre-existing one, or sa ...
````
```suggestion
This will open a Jupyter Notebook that contains the configurations for each of that Expectation Suite's Expectations in their own cells.  You can edit these cells and then run them to generate Expectations in a new Expectation Suite.  Once your edited version of the Expectations have been created in their own Expectation Suite, you can save that Expectation Suite over the pre-existing one, or save it as a new suite altogether.
```

talagluck commented on docs/guides/expectations/create_expectations_overview.md:
```diff
@@ -3,4 +3,67 @@ title: "Create Expectations: Overview"
+
+This command will list all the saved Expectation Suites in your Data Context.
+
+As long as you have a saved Expectation Suite to work with you'll be all set to move on to [Step 4: Validate Data.](../validation/validate_data_overview.md)
```
```suggestion
As long as you have a saved Expectation Suite with which to work, you'll be all set to move on to [Step 4: Validate Data.](../validation/validate_data_overview.md)
```

talagluck reviewed (commented):
Great content overall, and love the flow! Just a few minor nits, and a couple of tweaks for accuracy.

Rachel-Reverie committed:
Update docs/guides/expectations/create_expectations_overview.md

Co-authored-by: talagluck <tal@superconductive.com>

Rachel-Reverie committed:
Update docs/guides/expectations/create_expectations_overview.md

Co-authored-by: talagluck <tal@superconductive.com>

Rachel-Reverie committed:
Update docs/guides/expectations/create_expectations_overview.md

Co-authored-by: talagluck <tal@superconductive.com>

Rachel-Reverie committed:
Update docs/guides/expectations/create_expectations_overview.md

Co-authored-by: talagluck <tal@superconductive.com>

Rachel-Reverie committed:
Update docs/guides/expectations/create_expectations_overview.md

I'm fine with calling it the profiler workflow.  I'll make that change as well.

Co-authored-by: talagluck <tal@superconductive.com>

Rachel-Reverie committed:
Update docs/guides/expectations/create_expectations_overview.md

Co-authored-by: talagluck <tal@superconductive.com>

Rachel-Reverie committed:
Update docs/guides/expectations/create_expectations_overview.md

I think this needs to be "When following one of" but I'll make that edit after committing these changes.

Co-authored-by: talagluck <tal@superconductive.com>

Rachel-Reverie committed:
Update docs/guides/expectations/create_expectations_overview.md

Co-authored-by: talagluck <tal@superconductive.com>

Rachel-Reverie committed:
Changed "automatic" workflow to "Profiler" workflow.  Minor edits for grammar.

Rachel-Reverie committed:
changed "in depth" to "in-depth" where appropriate.
